pkg/api/api.go

```diff
@@ -252,7 +252,7 @@ type Ingester interface {
 	FlushHandler(http.ResponseWriter, *http.Request)
 	ShutdownHandler(http.ResponseWriter, *http.Request)
 	PrepareShutdownHandler(http.ResponseWriter, *http.Request)
-	PushWithCleanup(context.Context, *push.Request) error
+	PushWithCleanup(ctx context.Context, req *mimirpb.WriteRequest, cleanUp func()) error
```

Comment:
super-nit: other methods in this interface don't have parameter names, we should be consistent.